sycl/test-e2e/AtomicRef/atomic_memory_order_seq_cst.cpp

```diff
@@ -40,7 +37,7 @@ void check(queue &q, buffer<int, 2> &res_buf, size_t N_iters) {
   q.submit([&](handler &cgh) {
     auto res = res_buf.template get_access<access::mode::read>(cgh);
     auto checked =
-        checked_buf.template get_access<access::mode::discard_write>(cgh);
```

Comment:
Is it a temporary problem with `discard_write` caused by, e.g., some bug in GPU driver, or we should simply always use `write` instead of `discard_write` in such scenarios? If this is temporary problem in GPU driver, should we wait until GPU driver is updated on CI machines?

Reply:
I believe it is more of a misunderstanding of what `discard_write` does. It is now deprecated, but is the same as `write` and `no_init`. The problem here is that since we tell the device allocation that we don't care about the initial value of the memory, it is free to write back whatever value we don't write, while for this test we actually initialize the memory ourselves earlier.